components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1494,4 +1411,448 @@ private static Validator resourceToValidatorObject(
         }
         return validator;
     }
+
+    /**
+     * Get validators from the configuration store.
+     *
+     * @param tenantId Tenant Id
+     * @return List of Validator
+     * @throws ConfigurationManagementException Error when getting resource
+     */
+    public static List<Validator> getValidatorsFromConfigStore(int tenantId)
+            throws ConfigurationManagementException {
+
+        List<Validator> validators = new ArrayList<>();
+        Resources resources = CertValidationDataHolder.getInstance().getConfigurationManager()
+                .getResourcesByType(tenantId, VALIDATOR_RESOURCE_TYPE);
+        resources.getResources().forEach(resource -> validators
+                .add(resourceToValidatorObject(resource)));
+        return validators;
+    }
+
+    /**
+     * Get validator from the configuration store by name.
+     *
+     * @param tenantId Tenant Id
+     * @param name     Validator name
+     * @return Validator
+     * @throws CertificateValidationException Error when getting validator
+     */
+    public static Validator getValidatorFromConfigStoreByName(int tenantId, String name)
+            throws CertificateValidationException {
+
+        try {
+            org.wso2.carbon.identity.configuration.mgt.core.model.Resource resource =
+                    CertValidationDataHolder.getInstance().getConfigurationManager()
+                            .getResourceByTenantId(tenantId, VALIDATOR_RESOURCE_TYPE, name);
+            if (resource == null) {
+                return null;
+            }
+            return resourceToValidatorObject(resource);
```

Comment:
Can use `Optional.of(resourceToValidatorObject(resource)`, if we are using Optional<Validator> as the return type. Here its need to make sure `resourceToValidatorObject()` returns not null reference.

Reply:
ack

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/57/commits/52f92488d265780c25c0d23c5ec638ca00bd3f59